### DISCUSSION TITLE
How to get this style qr (circle)

### DISCUSSION THREAD

**Discussion by ugnius1994** [chillerlan/php-qrcode#226] [Q&A]
![qrround](https://github.com/chillerlan/php-qrcode/assets/46896856/96a5a21b-9e4c-4c3d-8ae8-ff8a5b80510f)

I tried     $options = new QROptions([
        'version'              => 5,
        'outputType'           => QRCode::OUTPUT_IMAGE_PNG,
        'eccLevel'             => QRCode::ECC_L,
        'drawCircularModules'  => true,
    ]);

**Comment by codemasher** [comment]
The settings seem fine to me - are you using the right version (v5-beta or dev-main)?

**Reply by ugnius1994** [reply]
It was 4.3v and updated to 5-beta but i get now
`Fatal error: Uncaught chillerlan\QRCode\Data\QRCodeDataException: code length overflow. (1040 > 864 bit) in`

**Reply by codemasher** [reply]
This happens because you set a fixed version number and the input is larger than this qr version can hold. You can either increase the version number or remove this setting entirely to enable version auto detection. See also: https://php-qrcode.readthedocs.io/en/main/Appendix/Terminology.html#version

**Reply by ugnius1994** [reply]
thanks, but i can't achieve same style qr :/.
And look what is difference between Qr codes.
![bad](https://github.com/chillerlan/php-qrcode/assets/46896856/759a3716-bb9d-4676-871e-f00cca097b55)

**Reply by codemasher** [reply]
The difference of the QR Codes between the versions is explained in the above link. In order to properly style the QR symbol, check out the example for the proper library version: https://github.com/chillerlan/php-qrcode/blob/main/examples/image.php
